services/QuillLMS/client/app/bundles/Staff/components/activityForm/topics.tsx

```diff
@@ -39,12 +39,34 @@ const Topics = ({ activity, createNewTopic, topicOptions, handleTopicsChange, })
   }
 
   function onChangeTopics(topicId) {
-    let newTopicIds = [...activity.topic_ids, topicId]
+    let newTopicIds = [...activity.topic_ids]
     const topic = topicOptions.find(t => t.id === topicId)
     const existingOption = getSelectedOptionForLevel(topic.level)
```

Comment:
I think the the variable names are making this hard to parse.

Only looking at the names, I would guess there are 3 different types: 
options: existingOption
nodes:  existingParent, existingGrandparent
topics: parentTopic, grandparentTopic

As I read through the logic, it seems like these are all just topic types (or some topic option types).  What about unifying the naming to reflect this:
existingTopic, existingParentTopic, existingGrandparentTopic, parentTopic, grandparentTopic
or
existingOption, existingParentOption, existingGrandparentOption, parentOption, grandparentOption